src/uct/ib/rdmacm/rdmacm_iface.c

```diff
@@ -0,0 +1,214 @@
+                                  const uct_device_addr_t *dev_addr,
+                                  const uct_iface_addr_t *iface_addr)
+{
+    /* Since the iface's address is the sockaddr, we already have the
+     * uct_md_is_sockaddr_accessible API call to check accessibility to
+     * this sockaddr */
+    return 1;
+}
+
+ucs_status_t uct_rdmacm_iface_get_address(uct_iface_h tl_iface, uct_iface_addr_t *iface_addr)
+{
+    uct_rdmacm_iface_t *iface = ucs_derived_of(tl_iface, uct_rdmacm_iface_t);
+    ucs_sock_addr_t *rdmacm_addr = (ucs_sock_addr_t *)iface_addr;
+
+    rdmacm_addr->addr    = rdma_get_local_addr(iface->cm_id);
+    rdmacm_addr->addrlen = sizeof(struct sockaddr);
+    return UCS_OK;
+}
+
+static uct_iface_ops_t uct_rdmacm_iface_ops = {
+    .ep_create_sockaddr       = UCS_CLASS_NEW_FUNC_NAME(uct_rdmacm_ep_t),
+    .ep_destroy               = UCS_CLASS_DELETE_FUNC_NAME(uct_rdmacm_ep_t),
+    .ep_flush                 = uct_base_ep_flush,
+    .ep_fence                 = uct_base_ep_fence,
+    .ep_pending_purge         = ucs_empty_function,
+    .iface_progress_enable    = (void*)ucs_empty_function_return_success,
+    .iface_progress_disable   = (void*)ucs_empty_function_return_success,
+    .iface_progress           = ucs_empty_function_return_zero,
+    .iface_flush              = uct_base_iface_flush,
+    .iface_fence              = uct_base_iface_fence,
+    .iface_close              = UCS_CLASS_DELETE_FUNC_NAME(uct_rdmacm_iface_t),
+    .iface_query              = uct_rdmacm_iface_query,
+    .iface_is_reachable       = uct_rdmacm_iface_is_reachable,
+    .iface_get_device_address = (void*)ucs_empty_function_return_success,
+    .iface_get_address        = uct_rdmacm_iface_get_address
+};
+
+static UCS_CLASS_INIT_FUNC(uct_rdmacm_iface_t, uct_md_h md, uct_worker_h worker,
+                           const uct_iface_params_t *params,
+                           const uct_iface_config_t *tl_config)
+{
+    uct_rdmacm_iface_config_t *config = ucs_derived_of(tl_config, uct_rdmacm_iface_config_t);
+    size_t ip_len = ucs_max(INET_ADDRSTRLEN, INET6_ADDRSTRLEN);
+    char *ip_str  = ucs_alloca(ip_len);
+    uct_rdmacm_md_t *rdmacm_md;
+    ucs_status_t status;
+
+    if (!(params->open_mode & UCT_IFACE_OPEN_MODE_SOCKADDR_SERVER) &&
+        !(params->open_mode & UCT_IFACE_OPEN_MODE_SOCKADDR_CLIENT)) {
+        ucs_fatal("Invalid open mode %zu", params->open_mode);
+    }
+
+    UCS_CLASS_CALL_SUPER_INIT(uct_base_iface_t, &uct_rdmacm_iface_ops, md, worker,
+                              params, tl_config UCS_STATS_ARG(params->stats_root)
+                              UCS_STATS_ARG(UCT_RDMACM_TL_NAME));
+
+    rdmacm_md = (uct_rdmacm_md_t *)self->super.md;
+    if (self->super.worker->async == NULL) {
+        ucs_error("rdmacm must have async != NULL");
+        return UCS_ERR_INVALID_PARAM;
+    }
+
+    self->cm_id                = NULL;
+    self->event_ch             = NULL;
```

Comment:
why need to set it here?
`cm_id` and  `event_ch` are set on lines `109` and `126` below